SCClassLibrary/JITLib/ProxySpace/NodeMap.sc

```diff
@@ -173,12 +174,58 @@ ProxyNodeMap : NodeMap {
 	}
 }
 
+
+// represents a subset of the output bus of a proxy
+// that can be mapped to a control.
+
+NodeProxyPartMap {
+	var <proxy, <indices;
+
+	*new { |proxy, indices=0|
+		^this.newCopyArgs(proxy, indices)
+	}
+
+	asControlInput {
+		var b = proxy.asControlInput;
+		var busMaps = if(b.isString) { [b] } { b };
+		^indices.asArray.collect { |i| busMaps.wrapAt(i) }.unbubble
+	}
+
+	asOSCArgEmbeddedArray { |array|
+		^this.asControlInput.asOSCArgEmbeddedArray(array)
+	}
+
+	wakeUpToBundle { |bundle, checkedAlready|
+		proxy.wakeUpToBundle(bundle, checkedAlready)
+	}
+
+	proxyNeedsWakeUp {
+		^true
+	}
+
+	proxyToWakeUp {
+		^proxy
+	}
+
+	removeChild { |obj|
+		proxy.removeChild(obj)
+	}
+
+	addChild { |obj|
+		proxy.addChild(obj)
+	}
+
+}
+
+
 + BusPlug {
 	asOSCArgEmbeddedArray { |array|
 		^this.asControlInput.asOSCArgEmbeddedArray(array)
 	}
 }
 
+
 + Object {
 	wakeUpToBundle {}
+	proxyNeedsWakeUp { ^false }
```

Comment:
Could there be another way of doing this instead of introducing a new member to Object?

Reply:
maybe we can ask if the object respondsTo(\proxyNeedsWakeUp) and then only implement this method in NodeProxy and NodeProxyPartMap?

Reply:
i mean, the previous code was 

`if (item.isKindOf(NodeProxy))`

now it would be 

`if (item.respondsTo(\proxyNeedsWakeUp) and: {item.proxyNeedsWakeUp})`

Reply:
that'a also not the best way (see above).

Btw. I wonder if `wakeUpToBundle` needs to be implemented in `Object`. But certainly, one could ask `respondsTo(\wakeUpToBundle)` – I just find that not such a good way of conveying intent yet.